apps/sms/services/js/device_storage/device_storage_shim.js

```diff
@@ -0,0 +1,59 @@
+/* global BridgeServiceMixin,
+          BroadcastChannel
+*/
+
+/* exported DeviceStorageShim */
+
+(function(exports) {
+'use strict';
+
+/**
+ * Name of the service for device storage API shim.
+ * @type {string}
+ */
+const SERVICE_NAME = 'device-storage-shim';
+
+const EVENTS = ['change'];
+
+const METHODS = ['lowDiskSpace'];
+
+var deviceStorage = null;
+
+var DeviceStorageShim = {
+  init(appInstanceId, realDeviceStorage) {
+    if (!realDeviceStorage) {
+      console.error('Invalid deviceStorage for shim initialization');
+      return;
+    }
+
+    deviceStorage = realDeviceStorage;
+
+    EVENTS.forEach((event) => {
+      deviceStorage.addEventListener(event, (evt) => {
+        // Workaround to avoid the duplicated events by inserting
+        // appInstanceId into event name in bridge mixin.
+        evt.appInstanceId = appInstanceId;
```

Comment:
Sorry, I'm not sure I understand why we need this, could you please clarify it a bit? I thought that using app-instance-bound-broadcast-channel is enough....